server/src/main/java/com/microsoft/java/bs/core/internal/gradle/GradleApiConnector.java

```diff
@@ -51,23 +51,40 @@ public GradleApiConnector(PreferenceManager preferenceManager) {
   }
 
   /**
-   * Get the Gradle version of the project.
+   * Extracts the GradleVersion for the given project.
+   *
+   * @param projectUri URI of the project to get the gradle version for.
+   * @return Gradle version of the project or empty upon failure.
    */
   public String getGradleVersion(URI projectUri) {
     try (ProjectConnection connection = getGradleConnector(projectUri).connect()) {
-      return getGradleVersion(connection);
+      return getBuildEnvironment(connection).getGradle().getGradleVersion();
     } catch (BuildException e) {
       LOGGER.severe("Failed to get Gradle version: " + e.getMessage());
       return "";
     }
   }
 
-  private String getGradleVersion(ProjectConnection connection) {
-    BuildEnvironment model = connection
+  /**
+   * Extracts the GradleJavaHome for the given project.
+   *
+   * @param projectUri URI of the project to get the gradle java home for.
+   * @return GradleJavaHome of the project or null upon failure.
+   */
+  public File getGradleJavaHome(URI projectUri) {
+    try (ProjectConnection connection = getGradleConnector(projectUri).connect()) {
+      return getBuildEnvironment(connection).getJava().getJavaHome();
+    } catch (BuildException e) {
+      LOGGER.severe("Failed to get Java version: " + e.getMessage());
+      return null;
+    }
+  }
+
+  private BuildEnvironment getBuildEnvironment(ProjectConnection connection) {
+    return connection
         .model(BuildEnvironment.class)
         .withArguments("--no-daemon")
```

Comment:
`--no-daemon` does not have effect when using tooling api

Reply:
Yes, I saw it get merged in #172. When this PR is ready for merge I will merge the last commit from the develop branch into this, resolving any merge conflicts. This should get fixed there.

Reply:
Or you can just rebase the PR based on the develop branch